Sources/RemoteMessaging/RemoteMessagingPercentileStoring.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  RemoteMessagingPercentileStoring.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public protocol RemoteMessagingPercentileStoring {
+    func percentile(forMessageId: String) -> Float
+}
+
+public class RemoteMessagingPercentileUserDefaultsStore: RemoteMessagingPercentileStoring {
```

Comment:
This is kept separate from the Core Data store on the iOS side, but let me know if you feel strongly.

One issue with this implementation is that percentage storage is unbounded, but I think the same is true of message storage. If I'm wrong about that, let me know and I'll look at updating this to clean up percentages.

Reply:
Correct, the message store is indeed unbounded and absolutely fine with me to keep this out of the Core Data store